quixstreams/models/serializers/avro.py

```diff
@@ -31,14 +40,30 @@ def __init__(
             Default - `False`
         :param disable_tuple_notation: If set to True, tuples will not be treated as a special case. Therefore, using a tuple to indicate the type of a record will not work.
             Default - `False`
+        :param schema_registry_config: If a Schema Registry configuration is provided, serialization is offloaded to Confluent's native AvroSerializer.
+            Default - `None`
         """
         self._schema = parse_schema(schema)
         self._strict = strict
         self._strict_allow_default = strict_allow_default
         self._disable_tuple_notation = disable_tuple_notation
+        self._schema_registry_serializer = None
+        if schema_registry_config:
+            client_config = schema_registry_config.client.as_dict(
+                plaintext_secrets=True
+            )
+            self._schema_registry_serializer = _AvroSerializer(
+                schema_registry_client=SchemaRegistryClient(client_config),
+                schema_str=json.dumps(schema),
+                conf=schema_registry_config.serialization.as_dict(),
+            )
 
     def __call__(self, value: Any, ctx: SerializationContext) -> bytes:
-        data = BytesIO()
+        if self._schema_registry_serializer:
```

Comment:
Can you replace it with `is None` check? 
It's more to the point and avoids calling the `__bool__` for the `_schema_registry_serializer` object.